statedb/treeupdate.go

```diff
@@ -129,6 +130,12 @@ func (u *TreeUpdate) Set(key, value []byte) error {
 	return u.tree.Set(u.tree.tx, key, value)
 }
 
+// Del removes a key-value from this tree.  `key` is the path of the leaf.
+func (u *TreeUpdate) Del(key []byte) error {
```

Comment:
I assume every time you delete a key the tree "rebalances". Do you think that so many deletes can impact on tx processing performance?

Reply:
The Del method is only used for the NoState() database currently.